### PR TITLE
fix case insensitive name exists query

### PR DESCRIPTION
**Pull request by dorianjp**
fix, previous name exists query was case sensitive, changed to insensitive

### CODE REVIEW

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]
Could you add a test case proving that the `lower` function is applied?

**Comment by dorianjp** [comment]
> Could you add a test case proving that the `lower` function is applied?

@AntoineAugusti good call, I fixed related test case and added another to demonstrate lower works in two scenarios with and without the same service ID.

File: tests/app/service/test_rest.py [hunk 1]
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
...
+                          ])
+def test_is_service_name_is_not_unique_returns_200_with_capitalization_while_unique_same_service(
+    admin_request,
+    notify_db,
+    notify_db_session,
+    name,
+    email_from
+):
+    service = create_service(service_name='unique', email_from='unique')
+    different_service_id = '111aa111-2222-bbbb-aaaa-111111111111'
```
Comment (AntoineAugusti) [inline tests/app/service/test_rest.py:2507]: You should be able to get a random UUID here with `uuid.uuid4()` (you can refactor in your other tests if you feel like it)

File: tests/app/service/test_rest.py [hunk 2]
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
...
+                         [("UNIQUE", "unique"),
+                          ])
+def test_is_service_name_is_not_unique_returns_200_with_capitalization_while_unique_same_service(
+    admin_request,
+    notify_db,
+    notify_db_session,
+    name,
+    email_from
+):
+    service = create_service(service_name='unique', email_from='unique')
```
Comment (AntoineAugusti) [inline tests/app/service/test_rest.py:2506]: I think you're not using the `email_from` from your parameters here and you should

File: tests/app/service/test_rest.py [hunk 3]
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
...
-def test_is_service_name_unique_returns_200_with_name_capitalized_or_punctuation_added(
+def test_is_service_name_unique_returns_200_with_punctuation_added(
     admin_request,
     notify_db,
     notify_db_session,
     name,
     email_from
 ):
-    create_service(service_name='unique', email_from='unique')
+    service = create_service(service_name='unique', email_from='unique')
```
Comment (AntoineAugusti) [inline tests/app/service/test_rest.py:2484]: I think you're not using the `email_from` from your parameters here and you should

File: tests/app/service/test_rest.py (hunk 1 above)
Comment (dorianjp) [reply tests/app/service/test_rest.py:2507]: yes, I just reused the existing hardcoded Id floating around in the tests prior

Comment (dorianjp) [reply tests/app/service/test_rest.py]: changed to uuid method

File: tests/app/service/test_rest.py (hunk 3 above)
Comment (dorianjp) [reply tests/app/service/test_rest.py:2483]: removed unused param

File: tests/app/service/test_rest.py (hunk 2 above)
Comment (dorianjp) [reply tests/app/service/test_rest.py:2504]: removed unused param

**Review by AntoineAugusti — APPROVED** [review]

**Comment by dorianjp** [comment]
[Decouple Email and Service Name](https://trello.com/c/yeyTL5es/158-decouple-email-and-service-name)

**Pull request merged**